### ISSUE TITLE
Already defines a member with the same parameter types

### ISSUE THREAD

**Issue by Xeevis** [christianhelle/refitter#269]
**Describe the bug**
When there are 2 similiar endpoints one with string in query and one in path, even though their endpoints, function and return type are different, they both get generated with exact same signature resulting in compiler error.

```csharp
[Get("/sale/categories")]
Task<CategoriesDto> CategoriesGet([Query, AliasAs("parent.id")] string parent_id);

[Get("/sale/categories/{categoryId}")]
Task<CategoryDto> CategoriesGet(string categoryId);
```

**OpenAPI Specifications**
[https://developer.allegro.pl/swagger.yaml](https://developer.allegro.pl/swagger.yaml)

**Comment by christianhelle** [comment]
@Xeevis I got something to work and I have tested that it also builds. Unfortunately, it's a bit of a breaking change as it involves changing some fundamental code for naming endpoint methods. So what I'll do is introduce more options so I don't introduce breaking changes

Can you try this file out and tell me if it works for you?

[Output.zip](https://github.com/christianhelle/refitter/files/13855321/Output.zip)

**Comment by christianhelle** [comment]
@all-contributors please add @Xeevis for bugs

**Comment by christianhelle** [comment]
@Xeevis Please keep me posted on whether the generated code works for you. I'll wait to merge in my fixes until I hear from you

**Comment by Xeevis** [comment]
> @Xeevis Please keep me posted on whether the generated code works for you. I'll wait to merge in my fixes until I hear from you

Hello. Thanks a lot for looking at it. Good thinking using the `operationId`, that's exactly what it's meant for.

![image](https://github.com/christianhelle/refitter/assets/5835044/7b9882b9-1374-411c-ac06-762c0dec59ec)

Just perhaps names could be normalized to meet C# naming conventions? (aka PascalCase and no underscores)

```csharp
// This is giving me goosebumps 
API.BadgeApplications_get_one("");
// Much better
API.BadgeApplicationsGetOne("");
```

**Comment by christianhelle** [comment]
@Xeevis Refitter is supposed to convert `_` to PascalCase so I must have done something wrong in the branch I have that generated the code. 

I'll fix whatever that problem was and I'll publish a release soon

**Comment by christianhelle** [comment]
@Xeevis  I merged the fix in and included it in the [v0.9.0](https://www.nuget.org/packages/Refitter/0.9.0) release

You need to pass in a few more configuration options because I didn't want to change the default behavior.

Here's the command I used to produce code:

```shell
refitter https://developer.allegro.pl/swagger.yaml --skip-default-additional-properties --operation-name-generator SingleClientFromOperationId
```

You need the `--skip-default-additional-properties` because NSwag generates a dictionary for properties that are not included in the OAS spec but might be transmitted over the wire anyway. This dictionary just happens to be named `AdditionalProperties` which conflicts with an existing `AdditionalProperties` definition in the `ShipmentDto` components/schema of the OAS document that you provided

NSwag generates the following if `--skip-default-additional-properties` is not specified:

```csharp
[JsonPropertyName("additionalProperties")]
public IDictionary<string, string> AdditionalProperties { get; set; }

private IDictionary<string, object> _additionalProperties;

[JsonExtensionData]
public IDictionary<string, object> AdditionalProperties
{
    get { return _additionalProperties ?? (_additionalProperties = new Dictionary<string, object>()); }
    set { _additionalProperties = value; }
}
```

make sure that you have the latest version of Refitter

```shell
dotnet tool update --global refitter
```

Hope this helps
